src/o365/teams/commands/report/teams-report-deviceusagedistributionusercounts.ts

```diff
@@ -39,18 +48,62 @@ class TeamsReportDeviceUsageDistributionUserCountsCommand extends GraphCommand {
     request
       .get(requestOptions)
       .then((res: any): void => {
-        cmd.log(res);
+        let content: string = '';
+
+        if (args.options.output && args.options.output.toLowerCase() === 'json') {
+          let reportdata = this.getJsonReport(res);
+          content = JSON.stringify(reportdata);
+
+          if (!args.options.outputFile) {
+            cmd.log(reportdata);
+          }
+        }
+        else {
+          content = res;
+
+          if (!args.options.outputFile) {
+            cmd.log(content);
+          }
+        }
+
+        if (args.options.outputFile) {
+          fs.writeFileSync(args.options.outputFile, content, 'utf8');
+          if (this.verbose) {
+            cmd.log(`File saved to path '${args.options.outputFile}'`);
+          }
+        }
 
         cb();
       }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
-}
+  }
+
+  private getJsonReport(res: string): any {
+    let rows = res.split('\n');
+    let jsonObj = [];
+    let headers = rows[0].split(',');
+
+    for (let i = 1; i < rows.length; i++) {
+      let data = rows[i].split(',');
+      let obj: any = {};
+      for (let j = 0; j < data.length; j++) {
+        obj[headers[j].trim()] = data[j].trim();
+      }
+      jsonObj.push(obj);
+    }
+
+    return jsonObj;
+  }
 
   public options(): CommandOption[] {
     const options: CommandOption[] = [
       {
         option: '-p, --period <period>',
         description: 'The length of time over which the report is aggregated. Supported values D7|D30|D90|D180',
         autocomplete: ['D7', 'D30', 'D90', 'D180']
+      },
+      {
+        option: '-f, --outputFile [outputFile]',
+        description: 'Path to the file where the upgrade report should be stored in'
       }
```

Comment:
Aren't we missing the `csv` output format which was included in the spec?

Reply:
In previous pull request Velin mentioned that we donot need to explcitly mention output format here.

Reply:
Removed the CSV format from the commands.